yarn-project/acir-simulator/src/avm/serialization/buffer_cursor.ts

```diff
@@ -0,0 +1,109 @@
+import { strict as assert } from 'assert';
+
+/*
+ * A Buffer-like class that automatically advances the position.
+ */
+export class BufferCursor {
+  constructor(private _buffer: Buffer, private _position: number = 0) {}
+
+  public position(): number {
+    return this._position;
+  }
```

Comment:
Nice! I should've pointed you to our `foundation` package earlier which has a lot of helper types, etc.

I found another [version of this class](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/foundation/src/serialize/buffer_reader.ts) (just for buffer-reading though). Thoughts on using that one or adding your functionality to that class?